src/pyscaffoldext/custom_extension/custom_extension.py

```diff
@@ -15,45 +19,67 @@ class CustomExtension(NoSkeleton):
     def activate(self, actions):
         actions = self.register(
                 actions,
-                self.add_custom_extension_structure,
+                add_custom_extension_structure,
                 after='define_structure')
+        actions = self.register(
+                actions,
+                set_pyscaffoldext_namespace,
+                after="add_custom_extension_structure"
+
+        )
         return self.register(
                 actions,
                 self.add_entry_point,
-                after='add_custom_extension_structure')
-
-    def add_custom_extension_structure(self, struct, opts):
-        custom_extension_file_content = extension(
-                self.get_class_name_from_opts(opts))
-        filename = "{}.py".format(opts["package"])
-        struct = helpers.ensure(struct, [opts["project"],
-                                         "src",
-                                         opts["package"],
-                                         filename],
-                                custom_extension_file_content,
-                                helpers.NO_OVERWRITE)
-
-        return struct, opts
+                after='set_pyscaffoldext_namespace')
 
     def add_entry_point(self, struct, opts):
         setup_cfg_content = struct[opts["project"]]["setup.cfg"][0]
-        config = configparser.ConfigParser()
+        config = ConfigUpdater()
         config.read_string(setup_cfg_content)
         config.remove_section("options.entry_points")
         config.add_section("options.entry_points")
         config.set("options.entry_points", "pyscaffold.cli",
-                   "{}={}.{}:{}".format(opts["package"],
-                                        opts["package"],
-                                        opts["package"],
-                                        self.get_class_name_from_opts(opts)))
-        buffer = io.StringIO()
-        config.write(buffer)
+                   "{}={}.{}.{}:{}".format(opts["package"],
+                                           opts["namespace"][-1],
+                                           opts["package"],
+                                           opts["package"],
```

Comment:
So the name of the extension module is also like the name of the package. I wonder if it wouldn't be better to just name it `extension.py` like it is done in our other external extensions. Personally I never name a Python module within a package like the package since a module should rather express with its name its purpose. The package name already tells you about what the name of the extension is, thus naming the module the same way is a repetition.

Reply:
Yes, it was among the modifications you proposed. I planned to go over them one by one. This one is just for the namespace thing. But I can modify this here, no probs.